### ISSUE TITLE
[Bug]: set_url without effect for instances of Line2D with linestyle 'none'

### ISSUE THREAD

**Issue by peunting** [matplotlib/matplotlib#28595]
### Bug summary

Bug summary
Using SVG-backend, set_url does nothing for for instances of Line2D with linestyle='none', whereas it works for other objects.
Related to closed issue: https://github.com/matplotlib/matplotlib/issues/17336

### Code for reproduction

```Python
from matplotlib import pyplot as plt

f = plt.figure()
s = plt.scatter([1, 2, 3], [4, 5, 6])
s.set_urls(['http://www.bbc.co.uk/news', 'http://www.google.com', None])
p = plt.plot([1, 3], [6, 5], linestyle='none', color='green', marker='x')
p[0].set_url('http://www.duckduckgo.com')

print(s.get_urls())
print(p[0].get_url())

import io
svg = io.StringIO()
f.savefig(svg, format='svg')
f.savefig('test.svg', format='svg')

assert "http://www.google.com" in svg.getvalue()
assert "http://www.duckduckgo.com" in svg.getvalue()
```

### Actual outcome

Exception has occurred: AssertionError
exception: no description
  File "/home/peunting/from matplotlib import pyplot as plt.py", line 18, in <module>
    assert "http://www.duckduckgo.com" in svg.getvalue()
AssertionError: 

### Expected outcome

Both http://www.google.com and http://www.duckduckgo.com should be present in the svg. When the svg is opened in a browser, the plot markers of the line plot should be clickable as well.

### Operating system

Rocky

### Matplotlib Version

3.8.2

### Matplotlib Backend

QtAgg

### Python version

3.10.12

### Installation

pip

**Comment by peunting** [comment]
I think in lines.py, line 812, the urls for the markers should be set:
https://github.com/matplotlib/matplotlib/blob/ed17e00da9b8119c0a01290ebcace64bc0120b84/lib/matplotlib/lines.py#L812

@QuLogic : Do you have an idea why it is not working?

**Comment by QuLogic** [comment]
Yes, that code means the backend receives the URL, but doesn't necessarily mean the backend does anything with it. For SVG, I don't see any handling of URLs for plain markers:
https://github.com/matplotlib/matplotlib/blob/e6f2b0c72be421f3b324a55fca1dbc367405cce5/lib/matplotlib/backends/backend_svg.py#L692-L729

**Comment by peunting** [comment]
wow, thanks for taking the issue and working on it!
